translator/gobgp.py

```diff
@@ -38,41 +55,50 @@ def _build_path(self, ip):
             )
         )
 
+        # Set the next hop to the correct value depending on IP family
         next_hop = Any()
         family = self._get_family(ip.ip.version)
-
-        if ip.ip.version == 6:
+        if family == 6:
             next_hop.Pack(
                 attribute_pb2.MpReachNLRIAttribute(
                     family=gobgp_pb2.Family(afi=family, safi=gobgp_pb2.Family.SAFI_UNICAST),
                     next_hops=["100::1"],
```

Comment:
I'm struggling to figure out how to do this well. Being that there's a next_hop ipv4 and ipv6, how do we set a default  if we don't know which they're passing? We could require both a next_hop_v4 and next_hop_v6 in the websocket message, but that seems like poor UX.

For example, I started going down this route, but obviously we'd never fall back to defaults then:
```
        if "next_hop" in event_data:
            if event_data["next_hop"].ip.version == 6:
                family = 6
                next_hop = event_data.get("next_hop", 100::1)
            else:
                family = 4
                next_hop = event_data.get("next_hop", 192.0.2.199)

```

So I could do the outer check and then set a default if `next_hop` wasn't set in `event_data`, but do I use the v4 or the v6 default? 

I'm going to sit and think on this a bit, and I welcome opinions. If you have a better way to handle it, I see the value in accepting next_hop this way, but as it stands I'm not sure it really buys us anything.

Reply:
Not necessary, but a nice to have would be to move all the defaults to the top, so it's easy to see them all and change them if anyone wants to.

```
DEFAULT_ASN = 65400
DEFAULT_COMMUNITY = 666
DEFAULT_V4_NEXTHOP = "192.0.2.199"
DEFAULT_V6_NEXTHOP = "100::1"
```

Then just do it in the same if block:

```
if family == 6:
   # Convert the single "next_hop" value to a list, so that in the future, 
   # we can more easily support someone passing several hops as a list.

   route_next_hops = list(event_data.get("next_hop", DEFAULT_V6_NEXTHOP))
   next_hop.Pack(..., next_hops=route_next_hops...)
```

Reply:
Followed this suggestion